# GWAS workflow for cotton nep traits
The scripts in this repository were used for GWAS and post-GWAS analyses of Cotton neps.

## Custom scripts
A customized Python script was used to eliminate duplicates.
Additionally, a customized Python script was used to calculate descriptive statistics.
R scripts from the original sources were modified for this dataset to create violin plots, LDHeatmaps, expression heatmaps, and to customize Manhattan plots and local plots.

## References
Browning, B.L., Browning, S.R., 2016. Genotype Imputation with Millions of Reference Samples. The American Journal of Human Genetics 98, 116–126. https://doi.org/10.1016/j.ajhg.2015.11.020

Cingolani, P., Platts, A., Wang, L.L., Coon, M., Nguyen, T., Wang, L., Land, S.J., Lu, X., Ruden, D.M., 2012. A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118 ; iso-2; iso-3. Fly 6, 80–92. https://doi.org/10.4161/fly.19695

Huang M, Liu X, Zhou Y, Summers RM, Zhang Z. BLINK: A package for the next level of genome-wide association studies with both individuals and markers in the millions. Gigascience. https://doi.org/10.1093/gigascience/giy154

Kolde R., 2025. pheatmap: Pretty Heatmaps. R package version1.0.13, https://github.com/raivokolde/pheatmap

Lipka A. E., Tian F., Wang Q., Peiffer J., Li M., et al., 2012 GAPIT: genome association and prediction integrated tool. Bioinformatics 28: 2397–2399, https://doi.org/10.1093/bioinformatics/bts444

Liu X., Huang M., Fan B., Buckler E. S., Zhang Z., 2016 Iterative Usage of Fixed and Random Effect Models for Powerful and Efficient Genome-Wide Association Studies. PLoS Genet. 12: e1005767. https://doi.org/10.1371/journal.pgen.1005767

Paria, S.S., Rahman, S.R., Adhikari, K., 2022. fastman: A fast algorithm for visualizing GWAS results using Manhattan and Q-Q plots. https://doi.org/10.1101/2022.04.19.488738

Shin, J.-H., Blay, S., Graham, J., McNeney, B., 2006. LDheatmap : An R Function for Graphical Display of Pairwise Linkage Disequilibria Between Single Nucleotide Polymorphisms. J. Stat. Soft. 16. https://doi.org/10.18637/jss.v016.c03

Wang J., Zhang Z., GAPIT Version 3: Boosting Power and Accuracy for Genomic Association and Prediction, Genomics, Proteomics & Bioinformatics (2021), doi: https://doi.org/10.1016/j.gpb.2021.08.005

Yin, L., Zhang, H., Tang, Z., Xu, J., Yin, D., Zhang, Z., Yuan, X., Zhu, M., Zhao, S., Li, X., Liu, X., 2021. rMVP: A Memory-Efficient, Visualization-Enhanced, and Parallel-Accelerated Tool for Genome-Wide Association Study. Genomics, Proteomics & Bioinformatics 19, 619–628. https://doi.org/10.1016/j.gpb.2020.10.007
